docs/Understanding-Makefile.rst

```diff
@@ -0,0 +1,316 @@
+Customizing the Makefiles from Symbiflow-examples For Your Own Designs
```

Comment:
I think that, rather than the Makefiles, which is just a way of automatically executing some commands (e.g. a bash script or python could be used instead), what would be important to describe are the various commands that are being run, e.g.:
- what does `symbiflow_synth` internally do, etc.?
- what are the arguments that are to be provided, and what do they mean?

I'd say that the important thing is to not focus on how to use Makefiles, but rather on what are the required commands to run the toolchain to completion. Possibly this

Reply:
On a side note, it would be good to use the naming of the file consistent with other doc files, and have something like `understanding-makefiles`. Note that, per the previous comment, the name of this file will need to change.

Reply:
Thanks for the suggestion. I think that documentation for how the tool chain commands work is very import, but I also think that instructions on how the Makefiles in symbiflow work and how to modify them for personal designs is important. I have just finished creating two separate sections within the documentation describing how to implement both of these methods and have pushed them to this PR. I believe that separate sections on both topics are beneficial for three reasons: 1) Users who are just starting with symbiflow/linux may be more comfortable with modifying existing makefiles then they would be to create there own scripts 2) The makefile documentation provides valuable insights that might prove useful for debugging issues encountered while building the provided examples 3) by separating instructions into two parts, users are empowered with the option to chose which method (ether makefile, python, or bash script) works best for them. Let me know what you think.